test/framework/src/main/java/org/elasticsearch/plugins/MockPluginsService.java

```diff
@@ -66,7 +66,7 @@ public MockPluginsService(Settings settings, Environment environment, Collection
             }
             pluginsLoaded.add(new LoadedPlugin(pluginInfo, plugin, pluginClass.getClassLoader(), ModuleLayer.boot()));
         }
-
+        loadExtensions(pluginsLoaded);
```

Comment:
this is to ensure that the test framework loads the extensions...it probably wasn't necessary until now.

Reply:
That's surprising consider we have quite a few SPI already. But for Security, I know we have a [test only constructor](https://github.com/elastic/elasticsearch/blob/295c6f06236418b28564b49e6c98314f52246e6e/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java#L531) that explicitly accepts a list of extensions. Maybe that's how other SPIs are handled as well.